### PR TITLE
[ECS] Add Code Checking In the Tests Directory and Implement Fixes For the New Code

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13                   |
| Bug fix?        | no                                                      |
| New feature?    | no                                                      |
| BC breaks?      | no                                                      |
| Deprecations?   | no |
| Related tickets |  https://github.com/Sylius/Sylius/pull/15532#discussion_r1400672110                     |
| License         | MIT                                                          |

To save developers' effort during the code review, we added coding standards checks into the unit tests directory as well.

### CODE REVIEW

**Review by diimpp — DISMISSED** [review]

**Review by Wojdylak — DISMISSED** [review]

**Comment by jakubtobiasz** [comment]
@Rafikooo can you solve conflicts 😅?

**Review by NoResponseMate — DISMISSED** [review]

File: phpstan.neon.dist [hunk 1]
```diff
@@ -42,3 +42,4 @@ parameters:
         - '/Class Symfony\\Component\\Security\\Core\\Authentication\\Token\\UsernamePasswordToken constructor invoked with 4 parameters\, 2\-3 required./'
         - '/Method Symfony\\Component\\Serializer\\Normalizer\\NormalizerInterface\:\:supportsNormalization\(\) invoked with 3 parameters\, 1\-2 required\./'
         - '/Symfony\\Component\\Serializer\\NameConverter\\NameConverterInterface::normalize\(\) invoked with 2 parameters, 1 required./'
+        - '/PHPDoc tag \@extends contains generic type Symfony\\Component\\Security\\Core\\User\\UserProviderInterface\<Symfony\\Component\\Security\\Core\\User\\UserInterface\> but interface Symfony\\Component\\Security\\Core\\User\\UserProviderInterface is not generic\./'
```
Comment (Wojdylak) [inline phpstan.neon.dist]: I don't know what is better https://github.com/Sylius/Sylius/blob/1.13/phpstan-baseline.neon#L5584

File: psalm.xml [hunk 2]
```diff
@@ -271,6 +271,11 @@
                 <referencedFunction name="Symfony\Component\PasswordHasher\PasswordHasherInterface::hash" />
             </errorLevel>
         </TooManyArguments>
+        <TooManyTemplateParams>
+            <errorLevel type="info">
+                <file name="src/Sylius/Bundle/UserBundle/Provider/UserProviderInterface.php" /> <!-- TODO: remove after dropping Symfony 5.4 -->
+            </errorLevel>
+        </TooManyTemplateParams>
```
Comment (Wojdylak) [inline psalm.xml]: Here the same https://github.com/Sylius/Sylius/blob/1.13/psalm.xml#L190 :thinking:

File: phpstan.neon.dist (hunk 1 above)
Comment (Rafikooo) [reply phpstan.neon.dist]: I have separated the static analysis changes into a separate PR. Let's continue the discussion here:
https://github.com/Sylius/Sylius/pull/15578

File: src/Sylius/Bundle/CoreBundle/Validator/Constraints/ChannelCodeCollectionValidator.php [hunk 3]
```diff
@@ -28,7 +28,7 @@ public function __construct(private ChannelRepositoryInterface $channelRepositor
     {
     }
 
-    /** @param null|array<array-key, mixed> $value */
+    /** @param array<array-key, mixed>|null $value */
```
Comment (diimpp) [inline src/Sylius/Bundle/CoreBundle/Validator/Constraints/ChannelCodeCollectionValidator.php]:
```suggestion
    /** @param array<array-key, mixed>|mixed $value */
```

Theoretically in can be anything? (not sure if it's a valid docblock though)

**Review by diimpp — DISMISSED** [review]

**Review by Wojdylak — DISMISSED** [review]

**Review by vvasiloi — DISMISSED** [review]

File: src/Sylius/Bundle/CoreBundle/Validator/Constraints/ChannelCodeCollectionValidator.php (hunk 3 above)
Comment (Rafikooo) [reply src/Sylius/Bundle/CoreBundle/Validator/Constraints/ChannelCodeCollectionValidator.php]: I set `/** @param mixed $value */` to conform to the parent class documentation

**Review by Wojdylak — APPROVED** [review]

**Review by NoResponseMate — APPROVED** [review]

**Comment by NoResponseMate** [comment]
Thank you, @Rafikooo!

**Pull request merged**